### PR TITLE
js_of_ocaml: fix jbuild (explicitly call bash)

### PR DESCRIPTION
**Pull request by ibor**
ocamlPackages.js_of_ocaml-3.0.0 currently fails to build because bash is not called properly.
The patch fixes jbuild to explicitly call generate_stubs.sh via bash.

BTW: Modifying generate_stubs.sh's shebang (#!/bin/bash -> #!/usr/bin/env bash ) did not help.

### CODE REVIEW

File: pkgs/development/tools/ocaml/js_of_ocaml/3.0.nix [hunk 1]
```diff
@@ -11,5 +11,7 @@ stdenv.mkDerivation rec {
 
 	propagatedBuildInputs = [ js_of_ocaml-compiler uchar ];
 
+  patches = [ ./jbuild.diff ];
```
Comment (orivej) [inline pkgs/development/tools/ocaml/js_of_ocaml/3.0.nix]: This is usually done with `patchShebangs` in the `postPatch` phase.

Comment (ibor) [reply pkgs/development/tools/ocaml/js_of_ocaml/3.0.nix]: Thank you. That works an is indeed much simpler:
https://github.com/NixOS/nixpkgs/compare/master...ibor:master

Do you need a new pull request?

Comment (orivej) [reply pkgs/development/tools/ocaml/js_of_ocaml/3.0.nix]: This pull request is fine (although its individual commits could be squashed together for the better), but I don't have the permission to merge yet. (I hope to get it soon.)

**Comment by peterhoeg** [comment]
In general, please squash the commits. 

Thank you for your contribution!

**Pull request merged**